Implement light/dark theme support

VedmaPranav opened the pull request:
This PR addresses Issue [#3](https://github.com/OWASP/Nest/issues/3)

Screenshot of the change:
<img width="1790" alt="Screenshot 2024-09-29 at 3 45 19 PM" src="https://github.com/user-attachments/assets/6d2fde29-3c2f-46eb-998f-2d37785c54e9">
<img width="1786" alt="Screenshot 2024-09-29 at 3 45 45 PM" src="https://github.com/user-attachments/assets/ca59c9fc-edad-4598-9f17-801dae9cbfce">


Please feel free to suggest different styling, or any other feedback

arkid15r reviewed (commented):
Hey @VedmaPranav thanks for the PR!

Could you format code using `make pre-commit` and update the PR?

VedmaPranav commented:
@arkid15r I ran the pre-commit checks and updated the PR

VedmaPranav commented:
I mistakenly merged the main branch into this branch, I need to undo the merge.

arkid15r commented:
> I mistakenly merged the main branch into this branch, I need to undo the merge.

So the pre-commit step of CI/CD is still failing. Could you address the pre-commit requirement issues?

arkid15r commented on backend/templates/base.html:
I noticed you didn't use the recommended bootstrap way to implement light/dark theme -- https://getbootstrap.com/docs/5.3/customize/color-modes/#enable-dark-mode

This looks like a more straight-forward solution and might not require a ton of extra CSS.
Do you think it's worth looking into the bootstrap recommended way and remove the extra complexity?

VedmaPranav replied:
@ arkid15r I suggest a combination of Bootstrap and some custom CSS. The bootstrap recommended adding a dark theme to the HTML tag element; however, since the page class has a background color set to #f4f6fc. So, when bootstrap dark theme is applied, we would see something as the following screenshot
<img width="1792" alt="image" src="https://github.com/user-attachments/assets/01b0c6be-f1b9-484e-aa89-4f4458e4a21a">
So, we must set the background color to None when the dark theme is applied. Also, the tooltips/github icons in the dark mode have a white border; we can remove this border through some custom CSS.  

app.css file also contains some CSS related to the light-dark slider.  Switching this slider to a button or something similar will lose the aesthetic of a toggle slider.

The following is a screenshot of how the page would look with the changes I suggested. let me know your thoughts
<img width="1792" alt="image" src="https://github.com/user-attachments/assets/a11d1331-adab-45d2-9408-ccd0da2f86e5">

arkid15r replied:
The second screenshot looks better. My suggestion was mostly about the extra work that could be avoided by using the bootstrap built-in stuff.

arkid15r commented on backend/static/img/light-bulb-2-16.png:
Can we use font-awesome icons for the style consistency?

VedmaPranav replied:
changed the icons to font-awesome icons

arkid15r replied:
I'm not that good at FE part, so my question here is would it be possible to avoid adding the actual images and use fa-moon/fa-sun/fa-lightbulb icons instead?

arkid15r reviewed (commented):
@VedmaPranav thanks addressing the pre-commit issue

Please find my comment below and let me know your thoughts

arkid15r reviewed (commented):
Thank you @VedmaPranav
Please push the new code when you have a chance.

VedmaPranav commented:
Hey @arkid15r 
I have removed any extra CSS that may not be used in the feature and changed the icons from images to font awesome like you suggested let me know if you need me to make anymore changes

arkid15r commented on backend/templates/base.html:
```diff
@@ -11,6 +11,7 @@
         </title>
         <link rel="stylesheet"
               href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0-beta3/css/all.min.css" />
+        <link rel="stylesheet" href="/static/css/app.css" />
```
I recommend using `{% static %}` tag here. Please see [example](https://github.com/OWASP/Nest/pull/15/files#diff-d448116430234e6d313e9295459f70cc2124cf7530d5f14039e023f2ae44668aR39).

arkid15r reviewed (commented):
This looks nice! 

Just a small comment regarding static path:

VedmaPranav commented:
@arkid15r I updated the base html file to use static tag

arkid15r reviewed (approved):
Great improvement! Thanks @VedmaPranav!
